Release/src/http/client/http_client_asio.cpp

```diff
@@ -864,12 +863,6 @@ class asio_context final : public request_context, public std::enable_shared_fro
             // Enforce HTTP connection keep alive (even for the old HTTP/1.0 protocol).
             request_stream << "Connection: Keep-Alive\r\n\r\n";
 
-            // Start connection timeout timer.
-            if (!ctx->m_timer.has_started())
-            {
-                ctx->m_timer.start();
-            }
-
```

Comment:
That is not needed anymore, as the conditional below should clearly distinguish the states of the timer:

a) The connection is reused (or an established proxy tunnel): In both cases the call to `ctx->write_request()` will eventually call `m_timer.start_communicating()`. Note that it is legal to call `start_communicating()` without a prior call to `start_connecting()` (e.g. when reusing an already established connection).

b) Otherwise, no connection is established yet, and a call to `m_timer.start_connecting()` is necessary.